packages/netlify/src/generators/setup-serverless/setup-serverless.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  addDependenciesToPackageJson,
+  convertNxGenerator,
+  formatFiles,
+  generateFiles,
+  GeneratorCallback,
+  joinPathFragments,
+  logger,
+  names,
+  readNxJson,
+  readProjectConfiguration,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
+import {
+  netlifyCliVersion,
+  netlifyFunctionVersion,
+} from '../../../utils/versions';
+import { SetupServerlessFunctionOptions } from './schema';
+
+const serverlessPlatformPackage = {
+  netlify: {
+    '@netlify/functions': netlifyFunctionVersion,
+    'netlify-cli': netlifyCliVersion,
+  },
+};
+function normalizeOptions(
+  tree: Tree,
+  setupOptions: SetupServerlessFunctionOptions
+) {
+  const project = setupOptions.project ?? readNxJson(tree).defaultProject;
+  const siteName = names(project).fileName;
+  return {
+    ...setupOptions,
+    project,
+    site: setupOptions.site ?? siteName,
+    lintTarget: setupOptions.lintTarget ?? 'lint',
+    buildTarget: setupOptions.buildTarget ?? 'build',
+    deployTarget: setupOptions.deployTarget ?? 'deploy',
+  };
+}
+
+function addServerlessFiles(
+  tree: Tree,
+  options: SetupServerlessFunctionOptions
+) {
+  const project = readProjectConfiguration(tree, options.project);
+  if (!options.project || !options.deployTarget) {
+    return;
+  }
+
+  switch (options.platform) {
+    case 'netlify':
+      if (tree.exists(joinPathFragments(project.root, 'netlify.toml'))) {
+        logger.info(
+          `Skipping setup since a netlify.toml already exists inside ${project.root}`
+        );
+      }
+      break;
+
+    default:
+      return;
+  }
+  const outputPath =
+    project.targets[`${options.buildTarget}`]?.options.outputPath;
+  generateFiles(
+    tree,
+    joinPathFragments(__dirname, `./files/${options.platform}`),
+    project.root,
+    {
+      tmpl: '',
+      app: project.sourceRoot,
+      outputPath,
+    }
+  );
+}
+
+function updateProjectConfig(
+  tree: Tree,
+  options: SetupServerlessFunctionOptions
+) {
+  const projectConfig = readProjectConfiguration(tree, options.project);
+
+  if (projectConfig) {
+    projectConfig.targets[`${options.buildTarget}`].options.bundle = true;
+    projectConfig.targets[`${options.deployTarget}`] = {
```

Comment:
Missing `executor: 'nx:run-commands`, or we need to specify the `command` rather than options (Nx will detect `command` field and infer `nx:run-commands`
 
We can also make it two different targets.

```
dev: {
  command: 'netlify dev'
},
deploy: {
  command: 'netlify deploy --prod-if-unlocked'
}
```

Reply:
I already added two different targets namely `development` and `production` 
I will add the executor